src/queues.c

```diff
@@ -353,13 +346,23 @@ void queues_update(bool fullscreen)
                 }
         }
 
+        int cur_displayed_limit;
+        if (settings.geometry.h == 0)
```

Comment:
The most important bit is the `displayed->length + waiting->length > settings.geometry.h`. This can't be done in `queues_init`. Putting this code into exactly this position is the actual fix for #467.